Jenkinsfile

```
@@ -3,79 +3,143 @@
+    docker.image('blueocean_build_env').inside {
+      withEnv(['GIT_COMMITTER_EMAIL=me@hatescake.com','GIT_COMMITTER_NAME=Hates','GIT_AUTHOR_NAME=Cake','GIT_AUTHOR_EMAIL=hates@cake.com']) {
+        try {
+          fullBuildSteps()                  
+        } catch(err) {
+          handleBuildFailure(err)
+        } finally {                      
+          deleteDir()          
+        }
+      }
+    }
+  }
+}
 
-      } catch(err) {
-        currentBuild.result = "FAILURE"
+def acceptanceTests() {
+  node {    
+    prepareEnvironment()
+    sh "./acceptance-tests/runner/scripts/start-selenium.sh"    
 
-        if (err.toString().contains('exit code 143')) {
-          currentBuild.result = "ABORTED"
-        }
-      } finally {
-        stage('Cleanup') {
+    docker.image('blueocean_build_env').inside("--net=container:blueo-selenium") {
+      withEnv(['GIT_COMMITTER_EMAIL=me@hatescake.com','GIT_COMMITTER_NAME=Hates','GIT_AUTHOR_NAME=Cake','GIT_AUTHOR_EMAIL=hates@cake.com']) {
+        try {
+          sh "mvn install -DskipTests"
+          runAcceptanceTests()
+          runWeeklyPermutationATH()
+        } catch(err) {
+          handleBuildFailure(err)
+        } finally {
           sh "${env.WORKSPACE}/acceptance-tests/runner/scripts/stop-selenium.sh"
-          sendhipchat()
-          deleteDir()
+          deleteDir()          
         }
       }
     }
   }
 }
 
 
-def sendhipchat() {
+
+/** 
+* The build steps for a build and unit test
+* Includes some npm sanity checks. 
+*/
+def fullBuildSteps() {
+  sh 'npm --prefix ./blueocean-core-js install'
+  sh 'npm --prefix ./blueocean-core-js run gulp'
+  sh "mvn install -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dmaven.test.failure.ignore -s settings.xml"
+  junit '**/target/surefire-reports/TEST-*.xml'
+  junit '**/reports/junit.xml'
+  archive '*/target/*.hpi'
+  
+  sh "node ./bin/checkdeps.js"
+  sh "node ./bin/checkshrinkwrap.js"
```

We should probably move these two node scripts to the front so the build fails fast for shrink issues